R/explore_text_columns.R

```diff
@@ -1,19 +1,311 @@
-#' Performs EDA of text features
+#' Performs EDA of text features. Works well with Rmarkdown.
+#' Its highly recommended to use code-chunk option results='asis'
+#' to make use of the markdown syntax
+#' - identifies text columns if not already specified
 #' - prints the summary statistics of character length
 #' - plots the distribution of character length
 #' - prints the summary statistics of word count
 #' - plots the distribution of word count
 #' - plots the word cloud
+#' - plots a bar plot of top 10 words
+#' - plot a word cloud of bigrams
+#' - plot a bar plot of top 10 bigrams
+#'
+#' @importFrom dplyr %>%
 #'
 #' @param df dataframe: dataset
-#' @param text_cols vector of numeric column names
+#' @param text_cols vector of text column names
 #'
-#' @return a list of plots
+#' @return a list of results and plots
+#' @export
 #'
 #' @examples
-#' \dontrun{
-#' explore_text_columns(df)
-#' }
-explore_text_columns <- function(df, text_cols=NULL){
+#' explore_text_columns(cars)
+explore_text_columns <- function(df, text_cols=vector(mode='character')) {
+
+  results <- list()
+  word_count <- frequen <- NULL
+
+  if (!is.character(text_cols)) {
+    stop("text_cols is not passed as a character vector")
+  }
+
+  if (!is.data.frame(df)) {
+    stop("The df passed is not a Data Frame")
+  }
+
+  if(length(text_cols) == 0) {
+    for (col in colnames(df %>% dplyr::select_if(is.character))){
+      if (df %>% dplyr::select(col) %>%
+          dplyr::n_distinct(col)/nrow(df) > 0.75){
+        if ((stats::median(df %>%
+                    dplyr::mutate(char_len = stringr::str_length(!!dplyr::sym(col))) %>%
+                    dplyr::pull(char_len))) > 5){
+          text_cols <- append(text_cols, col)
+        }
+      }
+    }
+    if(length(text_cols) == 0) {
+      cat("\n")
+      cat("\n")
+      cat("Could not identify any text column. Please pass the text column(s) when calling the function")
+      cat("\n")
+      cat("\n")
+      return(results)
+    }
+    else {
+      cat("\n")
+      cat("\n")
+      cat("Identified the following as text columns:", text_cols)
+      cat("\n")
+      cat("\n")
+    }
+  } else {
+    for (col in text_cols){
+      if (!(col %in% colnames(df))){
+        stop(paste0(col, " passed in text_cols is not a column in the dataframe"))
+      }
+      if (!is.character(df)){
```

Comment:
was this meant to be !is.character(col) instead of !is.character(df)?

Reply:
Good spot!
I've addressed this is in this [commit](https://github.com/UBC-MDS/datascience.eda.R/pull/15/commits/3ca7a07b6426286a79a5b8b44515395ebef1f3ff)